src/cfn/index.ts

```diff
@@ -1178,8 +1178,13 @@ abstract class AbstractCloudFormationStackCommand {
   protected watchStackEvents: boolean = true;
 
   constructor(readonly argv: GenericCLIArguments, readonly stackArgs: StackArgs) {
-    // note, this.region is set in _setup after the cal to configureAWS
-    this.profile = this.argv.profile || this.stackArgs.Profile;// tslint:disable-line
+    // region, profile, and assumeRoleArn are the only used for cli output here
+    // configureAWS is called by loadStackArgs prior to this constructor
+    // TODO We should cleanup / dry-up the resolution rules for these.
+    this.region = def(getCurrentAWSRegion(), this.argv.region || this.stackArgs.Region);
```

Comment:
What do you think about making `def` a varargs function (at that point it could basically just be `_.find`)? Using it on the `this.profile` assignment may make it cleaner as well.

Reply:
Let's follow up with that. I'm in favour of just going directly with `_.find` throughout.

Reply:
I think that `def` has value in communicating the intent – I don't think I would want to see it fully replaced with `_.find`.